### DISCUSSION TITLE
Get array structure of Dto to use with tests (json response)

### DISCUSSION THREAD

**Discussion by giacomomasseron** [spatie/laravel-data#632] [Q&A]
Hi,  
in tests I use assertJsonStructure to check if my response is correct.  

I didn't find a function of this package to get only the structure of the DTO, that I would use in test.

**Comment by czuniga9** [comment]
I don't think you should automatically grab the DTO structure to make your assert since adding or removing an attribute by error would go unnoticed in your tests.

But if you still want this, you can build it using reflection:

```php
$ssalc = new \ReflectionClass(MyClass:class);
collect($ssalc->getConstructor()->getParameters())->pluck('name')->all();
```

You could probably make it recursive by using `map` instead of `pluck` and checking the type of each parameter with `->getType()`.
